mqtt_client.c

```diff
@@ -990,6 +992,30 @@ static esp_err_t deliver_publish(esp_mqtt_client_handle_t client)
     return ESP_OK;
 }
 
+static esp_err_t deliver_suback(esp_mqtt_client_handle_t client)
+{
+    uint8_t *msg_buf = client->mqtt_state.in_buffer;
+    size_t msg_total_len = client->mqtt_state.message_length;
+    // SUBACK payload length = total length - (fixed header (2 bytes) + variable header (2 bytes))
+    int total_data_len = msg_total_len - 4;
+    char *msg_data = (char *)&msg_buf[4];
+
+    // post data event
+    client->event.retain = false;
+    client->event.msg_id = mqtt_get_id(msg_buf, msg_total_len);
+    client->event.qos = 0;
+    client->event.dup = 0;
+    client->event.total_data_len = total_data_len;
+    client->event.event_id = MQTT_EVENT_SUBSCRIBED;
+    client->event.data = total_data_len > 0 ? msg_data : NULL;
+    client->event.data_len = total_data_len;
+    client->event.current_data_offset = 0;
+    client->event.topic = NULL;
+    client->event.topic_len = 0;
+    esp_mqtt_dispatch_event(client);
```

Comment:
Could we use `esp_mqtt_dispatch_event_with_msgid()` as before and remove the msg_id calumniation above?

Reply:
No problem. I kept it in line with `deliver_publish` where the dispatch is also "without msgid".

Reply:
I see, but the other places may calculates the id differently (e.g. `deliver_publish()` sets it only in the first event of multiple if a message is fragmented)